crates/runtime/src/component/dataset/acceleration.rs

```diff
@@ -300,6 +307,17 @@ impl Acceleration {
         self.on_conflict = on_conflict;
         self
     }
+
+    pub fn partition_by_expressions(
+        &self,
+        ctx: &SessionContext,
+        df_schema: &DFSchema,
+    ) -> Result<Vec<Expr>, DataFusionError> {
+        self.partition_by
+            .iter()
+            .map(|sql| ctx.parse_sql_expr(sql, df_schema))
```

Comment:
This will need to enforce the restrictions on partition expressions from the spec, i.e. exactly one column reference, etc.

Reply:
added an `expression` module that builds validation criteria and does this conversion.